--- a/src/plugins/platforms/offscreen/qoffscreenintegration.cpp
+++ b/src/plugins/platforms/offscreen/qoffscreenintegration.cpp
@@ -6,7 +6,9 @@
 #include "qoffscreencommon.h"
 
 #if defined(Q_OS_UNIX)
+#if !defined(Q_OS_WASM)
 #include <QtGui/private/qgenericunixeventdispatcher_p.h>
+#endif
 #if defined(Q_OS_MAC)
 #include <qpa/qplatformfontdatabase.h>
 #include <QtGui/private/qcoretextfontdatabase_p.h>
@@ -341,7 +343,7 @@ QPlatformBackingStore *QOffscreenIntegration::createPlatformBackingStore(QWindow
 
 QAbstractEventDispatcher *QOffscreenIntegration::createEventDispatcher() const
 {
-#if defined(Q_OS_UNIX)
+#if defined(Q_OS_UNIX) && !defined(Q_OS_WASM)
     return createUnixEventDispatcher();
 #elif defined(Q_OS_WIN)
     return new QOffscreenEventDispatcher<QEventDispatcherWin32>();
